### ISSUE TITLE
Unresolved resource dependencies [DefaultCloudfrontDistribution] in the Outputs block of the template:

### ISSUE THREAD

**Issue by scconroy** [awslabs/aws-api-gateway-developer-portal#216]
When attempting to deploy from SAR (I presume the latest version. Attempted to launch a couple minutes ago.): 

https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:563878140293:applications~api-gateway-dev-portal

I'm getting this error when using a custom domain name:

"Unresolved resource dependencies [DefaultCloudfrontDistribution] in the Outputs block of the template"

Looking at the template, and the "DefaultCloudfrontDistribution" resource, there is a condition on the resource that's missing that causes it to not be created when a custom domain is used (when CustomDomainName and CustomDomainNameAcmCertArn are not null in the parameters).

Adding a "NoCustomDomainName" condition to WebsiteURL under outputs fixes this error (so, skipping it if there's no custom domain name, same as the missing resource "DefaultCloudfrontDistribution"). It doesn't seem like the output is necessary if a custom domain is used, so there may be a better workaround...

Working WebsiteURL Output:

```
WebsiteURL:
    Condition: NoCustomDomainName
    Description: URL for website hosted on S3
    Value:
      Fn::Join:
      - ''
      - - https://
        - Fn::GetAtt:
          - DefaultCloudfrontDistribution
          - DomainName
```

**Comment by Trial-In-Error** [comment]
Ah! Thanks for the heads up; I'll have a fix for this out very soon.

**Comment by Trial-In-Error** [comment]
#217 should fix the issue. Going to manually verify, then get it published to SAR. :)

**Comment by scconroy** [comment]
Fantastic, looking forward to trying it out myself after it's merged. Thank you!

**Comment by Trial-In-Error** [comment]
Ok! Fixed in v2.3.3 (live on master) and in SAR. Let me know if you have any other issues! :)
